src/hawkbit-client.c

```diff
@@ -303,20 +324,14 @@ static gint rest_request(enum HTTPMethod method, const gchar* url, JsonBuilder*
                                 g_debug("Failed to parse JSON response body. status: %ld", http_code);
                         }
                 }
-        } else if (res == CURLE_OPERATION_TIMEDOUT) {
-                // libcurl was able to complete a TCP connection to the origin server, but did not receive a timely HTTP response.
-                http_code = 524;
-                g_set_error(error,
-                            1,                    // error domain
-                            http_code,
-                            "HTTP request timed out: %s",
-                            curl_easy_strerror(res));
-        } else {
-                g_set_error(error,
-                            1,                    // error domain
-                            http_code,
-                            "HTTP request failed: %s",
+        }
+        if (res != CURLE_OK) {
```

Comment:
When `res` becomes FALSE, so we have any guarantee on http_code? Otherwise funtions will still return error indicator and set GError incosistently.

Reply:
I don't understand this. In case the curl operation does not succeed (`!= CURLE_OK`), the appropriate RHU_HAWKBIT_CLIENT_CURL_ERROR will be set and the functions returns FALSE. Only if `res` is `CURLE_OK` the `http_code` will be checked. Or am I missing something here?

Reply:
Ok, this behavior is changed in a later commit. When writing this note the method did still say `return http_code`.
(refer https://github.com/rauc/rauc-hawkbit-updater/pull/77/commits/a57e14ad4ed3878b809563907059d8a7e24dde2c)
For this commit that only claims to "introduce curl error domain", it should be fine.